File: restore.py
import re

with open('app_old.js', 'r', encoding='utf-16-le') as f:
    old_content = f.read()

with open('app.js', 'r', encoding='utf-8') as f:
    new_content = f.read()

# Find the block between setupEventListeners and setupCountryDropdown
match = re.search(r'(function setupEventListeners\(\) \{.*?\n\})\s*function setupCountryDropdown', old_content, re.DOTALL)
if match:
    func_block = match.group(1)
    
    # Insert before setupCountryDropdown in app.js
    if 'function setupCountryDropdown' in new_content:
        updated = new_content.replace('function setupCountryDropdown', func_block + '\n\nfunction setupCountryDropdown')
        with open('app.js', 'w', encoding='utf-8') as f:
            f.write(updated)
        print("Successfully restored setupEventListeners and setupYoutubeApiKeyModal.")
    else:
        print("Could not find setupCountryDropdown in app.js.")
else:
    print("Could not extract functions from app_old.js.")

File: restore.js
const fs = require('fs');
const oldApp = fs.readFileSync('app_old.js', 'utf16le');
const currentApp = fs.readFileSync('app.js', 'utf8');

const startMarker = 'function setupEventListeners() {';
const endMarker = 'function setupCountryDropdown() {';

const startIdx = oldApp.indexOf(startMarker);
const endIdx = oldApp.indexOf(endMarker);

if (startIdx === -1 || endIdx === -1) {
  console.log("Could not find boundaries in app_old.js");
  console.log("startIdx:", startIdx, "endIdx:", endIdx);
  process.exit(1);
}

const functionBody = oldApp.substring(startIdx, endIdx);

const insertTarget = 'function setupCountryDropdown() {';
const targetIdx = currentApp.indexOf(insertTarget);

if (targetIdx === -1) {
  console.log("Could not find target in app.js");
  process.exit(1);
}

const newApp = currentApp.substring(0, targetIdx) + functionBody + currentApp.substring(targetIdx);
fs.writeFileSync('app.js', newApp);
console.log("Successfully restored missing setup functions");
